libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -268,8 +268,137 @@ string IRGenerator::generateFunction(FunctionDefinition const& _function)
 			retParams += m_context.addLocalVariable(*varDecl).stackSlots();
 			retInit += generateInitialAssignment(*varDecl);
 		}
+
+		t("retParams", joinHumanReadable(retParams));
+		t("retInit", retInit);
```

Comment:
Maybe not needed in case we have modifiers (this is done in `generateFunctionWithModifierInner`)? In that case, can be moved to `<body>` for empty modifiers case, or something like:
```suggestion
		t("retInit", _function.modifiers().empty() ? retInit);
```

Reply:
The return variables are always initialized here. If we have modifiers, they are passed in to the next modifier and eventually to the inner function. There they are used and then returned again. This is needed so that we initialize the return variables even if `_` is not executed. The return variables are passed in so that they are only initialized once.